client-app/app/controllers/index.js

```diff
@@ -171,4 +182,60 @@ export default class IndexController extends Controller {
 
     debounce(this, this.doSearch, term, 250);
   }
+
+  @action
+  buildGroupingRegexFromSelectedRows() {
+    this.buildingGroupingRegex = true;
+  }
+
+  @action
+  async createGroupingRegexFromSelectedRows() {
+    if (this.rowMessagesForGroupingRegex.length < 2) {
+      // eslint-disable-next-line no-alert
+      return alert(
+        "You must select at least 2 rows to create a grouping regex"
+      );
+    } else {
+      const match = this.findLongestMatchingSubstring(
+        this.rowMessagesForGroupingRegex
+      );
```

Comment:
Once we have the longest matching prefix, we need to regex-escape it and add `^` to the beginning. (otherwise, log messages with characters like `.` in them will cause issues)

We could copy/paste [this escape function from core](https://github.com/discourse/discourse/blob/d5d0bab19d5b807e3068f97a5fc085fb4aa847ff/app/assets/javascripts/discourse-common/addon/utils/escape-regexp.js#L1-L3).